src/api/user.ts

```diff
@@ -1,14 +1,66 @@
 import axios from 'axios';
+
 import {
   ChangePasswordPropsType,
   EditProfilePropsType,
+  LoginPropsType,
   RegisterPropsType,
 } from '../schemas';
 import { User } from '../types';
+import { createFormData } from '../utils';
+
 import { ENDPOINTS } from './constants';
+import { getAccessToken, getUserToken, setToken } from '.';
 
 type UserId = string | number;
 
+export const registerUser = async (credentials: RegisterPropsType) => {
+  const { avatar, ...rest } = credentials;
+  const formData = createFormData({ data: rest });
+  avatar.length && formData.append('avatar', avatar[0]);
+
+  /// @ts-expect-error In fact formData is RegisterPropsType object
+  await createUser(formData);
+
+  const user = await loginUser({
+    email: rest.email,
+    password: rest.password,
+  });
+
+  return user;
+};
+
+export const loginUser = async ({ email, password }: LoginPropsType) => {
+  const {
+    access,
+    refresh,
+    user_id: id,
+  } = await getUserToken({
+    email,
+    password,
+  });
+  setToken(access);
+
+  const user = await getUserById(id);
+
+  return { user, token: { refresh, id } };
+};
+
+export const refreshUser = async ({
+  id,
+  refresh,
+}: {
+  id: UserId;
+  refresh: string;
+}) => {
+  const { access } = await getAccessToken(refresh);
+  setToken(access);
+
+  const user = await getUserById(id);
+
+  return { user, token: { refresh, id } };
+};
```

Comment:
I meant write this in thunk. We have different api functions with use inside asyncThunk.